src/evaluation/Run.jl

```diff
@@ -406,6 +422,54 @@ end
 update_save_run!(session::ServerSession, notebook::Notebook, cell::Cell; kwargs...) = update_save_run!(session, notebook, [cell]; kwargs...)
 update_run!(args...; kwargs...) = update_save_run!(args...; save=false, kwargs...)
 
+
+function cells_to_disable_to_resolve_multiple_defs(old::NotebookTopology, new::NotebookTopology, cells::Vector{Cell})::Dict{Cell,Any}
+	to_disable = Dict{Cell,Any}()
+	
+	for cell in cells
+		new_node = new.nodes[cell]
+		
+		fellow_assigners_old = filter!(c -> !is_disabled(old, c), where_assigned(old, new_node))
+		fellow_assigners_new = filter!(c -> !is_disabled(new, c), where_assigned(new, new_node))
+		
+		if length(fellow_assigners_new) > length(fellow_assigners_old)
+			other_definers = setdiff(fellow_assigners_new, (cell,))
+			
+			@debug "Solving multiple defs" cell.cell_id cell_id.(other_definers) disjoint(cells, other_definers)
+			if (
+				# we want cell to be the only element of cells that defines this varialbe, i.e. all other definers must have been created previously
+				disjoint(cells, other_definers) &&
+				
+				# all fellow cells (including the current cell) should meet the following criteria:
+				all(fellow_assigners_new) do c
+					node = new.nodes[c]
+					
+					# the cell defines only one variable. for more than one, we might confuse the user, or disable more things than we want to.
+					length(node.definitions) == 1 &&
+					# avoid something like `x = x + 1`
+					disjoint(node.references, node.definitions) &&
+					# no function definitions
+					isempty(node.funcdefs_without_signatures) &&
+					# no macro calls					
+					isempty(node.macrocalls)
```

Comment:
```suggestion
					# for more than one, we might confuse the user, or disable more things than we want to.
					only_one_definition = length(node.definitions) == 1
					no_self_reference = disjoint(node.references, node.definitions)
					no_function_defs = isempty(node.funcdefs_without_signatures)			
					no_macro_calls = isempty(node.macrocalls)
					return only_one_definition && no_self_reference && no_function_defs && no_macro_calls
```
Code is generally safer than comments because comments tend to become outdated and are often not read

Reply:
I went for a middle ground between the two, and I also removed the confusing `disjoint(cells, other_definers) &&` and just used two `if` statements. Thanks!